src/transformers/modeling_utils.py

```diff
@@ -371,13 +371,13 @@ def shard_checkpoint(
             storage_id = id_tensor_storage(weight)
 
         # If a `weight` shares the same underlying storage as another tensor, we put `weight` in the same `block`
-        if storage_id in storage_id_to_block:
+        # unless if is a 'meta' parameter tensor
+        if storage_id in storage_id_to_block and weight.device != torch.device("meta"):
```

Comment:
What happens with we don't do that ? Do all meta tensors end up in the same block ?